### PR TITLE
Add setter and getter for previous_loss_values_dump_amount

### PR DESCRIPTION
**Pull request by reunanen**
Wanted to increase `previous_loss_values_dump_amount` in my application.

### CODE REVIEW

**Comment by davisking** [comment]
Hmm, was it terminating too early?  I've done a whole lot of testing and the current settings seemed very stable.  Although you never know.  But maybe there is some other problem here.

**Comment by reunanen** [comment]
> Hmm, was it terminating too early?

Yes. Once the learning rate dropped from the initial 0.1, it quickly dropped all the way to some 1e-6. Cranking the number up to 2000 appeared to fix the problem.

> But maybe there is some other problem here.

Could be. I was using quite small mini-batches (size 10 or so); maybe that has something to do with it.

**Comment by davisking** [comment]
Huh.  And when you set it to 2000 there was further decrease in the loss eventually right?  It didn't basically just grind at the same loss?

**Comment by reunanen** [comment]
Correct. I bet 2000 was overkill, though.

**Comment by davisking** [comment]
Hmm.  Damn.  I was hoping to have that parameter set to a value that never needed to be fiddled with.  All the things I've tested so far have worked as expected with the default value.  Can you run your code a few times with different values and see what happens?  How far off from 400 is needed?  Is it really because of the mini-batch size?

**Comment by reunanen** [comment]
Ok – I'll try to. It'll take me a while to run these experiments, though.

**Comment by davisking** [comment]
Thanks.  I almost don't want to have this option if it doesn't work all the time.  It does make training a lot faster in some cases, but that's not so helpful if it can't be relied upon.

**Comment by reunanen** [comment]
Based on quick tests, it does look like: the smaller the mini-batch size, the more prone the learning process is to this problem (given the default value 400).

Let's assume you used a mini-batch size of around 80 most of the time. So for you, 400 steps * 80 samples per mini-batch = 32k samples. So, how about:

`previous_loss_values_dump_amount` = 32000 / mini-batch size

Just a random thought, really...

**Comment by davisking** [comment]
Yeah, let me think about it.  We want to know the answer to "how many to keep so that the probability of false alarm stays below X?".  I'll see if I can figure out the right answer to that.  Maybe it's divide by N or sqrt(N) or something more complex.

**Comment by davisking** [comment]
I just pushed a change that hopefully fixes this and removes the need for a user settable variable.  Can you pull the code and see if it fixes your issue?

**Comment by reunanen** [comment]
> I just pushed a change that hopefully fixes this and removes the need for a user settable variable.  Can you pull the code and see if it fixes your issue?

Thanks, I'll try it at some point.

**Comment by reunanen** [comment]
Looks like your fix is doing essentially the same thing as my application code, i.e. increasing the dump-amount relative to iterations-without-progress-threshold.

**Comment by davisking** [comment]
Yep, seems to give good results in my testing.  Thanks for reporting this issue.
